From e3ad09b001b6c0a643c4ddc5131ff123ea3fea58 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 25 Aug 2020 17:54:05 +0000
Subject: [PATCH 1/1] efi_loader: log messages for bootefi command

Write log messages when booting via the bootefi command to allow tracking
on the syslog server. Example messages are

    Booting /snp.efi

or

    Booting /MemoryMapped(0x0,0x4fe00000,0x35a40)
    Loading image failed

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 cmd/bootefi.c | 12 +++++++++---
 1 file changed, 9 insertions(+), 3 deletions(-)

diff --git a/cmd/bootefi.c b/cmd/bootefi.c
index fbfed54e85..8333e7b3f2 100644
--- a/cmd/bootefi.c
+++ b/cmd/bootefi.c
@@ -429,7 +429,9 @@ efi_status_t efi_run_image(void *source_buffer, efi_uintn_t source_size)
 {
 	efi_handle_t mem_handle = NULL, handle;
 	struct efi_device_path *file_path = NULL;
+	struct efi_device_path *msg_path;
 	efi_status_t ret;
+	u16 *load_options;
 
 	if (!bootefi_device_path || !bootefi_image_path) {
 		/*
@@ -452,17 +454,21 @@ efi_status_t efi_run_image(void *source_buffer, efi_uintn_t source_size)
 				       file_path);
 		if (ret != EFI_SUCCESS)
 			goto out;
+		msg_path = file_path;
 	} else {
 		file_path = efi_dp_append(bootefi_device_path,
 					  bootefi_image_path);
+		msg_path = bootefi_image_path;
 	}
 
+	log_info("Booting %pD\n", msg_path);
+
 	ret = EFI_CALL(efi_load_image(false, efi_root, file_path, source_buffer,
 				      source_size, &handle));
-	if (ret != EFI_SUCCESS)
+	if (ret != EFI_SUCCESS) {
+		log_err("Loading image failed\n");
 		goto out;
-
-	u16 *load_options;
+	}
 
 	/* Transfer environment variable as load options */
 	ret = efi_env_set_load_options(handle, "bootargs", &load_options);
-- 
2.20.1

